### PR TITLE
[skip ci] dashboard: support changing grafana admin pass

### PR DESCRIPTION
**Pull request by guits**
This commit adds the grafana admin password changing support.
    
Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1962629
    
Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-collocation

**Comment by guits** [comment]
jenkins test centos-container-collocation

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 1]
```diff
@@ -171,43 +171,69 @@
...
 
-- name: set alertmanager host
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-alertmanager-api-host http://{{ grafana_server_addrs | first }}:{{ alertmanager_port }}"
-  delegate_to: "{{ groups[mon_group_name][0] }}"
-  run_once: true
-  changed_when: false
+- name: set_fact prometheus_api_host, alertmanager_api_host, grafana_api_url
+  set_fact:
+    prometheus_api_host: "{{ dashboard_protocol }}://{{ prometheus_frontend_vip if prometheus_frontend_vip is defined and prometheus_frontend_vip | length > 0 else monitoring_server_addrs | first }}:{{ prometheus_port }}"
+    alertmanager_api_host: "{{ dashboard_protocol }}://{{ alertmanager_frontend_vip if alertmanager_frontend_vip is defined and alertmanager_frontend_vip | length > 0 else monitoring_server_addrs | first }}:{{ alertmanager_port }}"
```
Comment (dsavineau) [inline roles/ceph-dashboard/tasks/configure_dashboard.yml:177]: prometheus and alertmanager doesn't support https so you can't reuse the `dashboard_protocol` variable here

Comment (guits) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml:177]: maggle, this review comes at the right time :grinning: 
:+1:

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-update

File: roles/ceph-dashboard/tasks/configure_dashboard.yml (hunk 1 above)
Comment (fmount) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml]: @dsavineau ++ right, prometheus and alertmanager are left w/ http, in OSP for tls-everywhere we're using the approach described in [1] (terminating the ssl on the haproxy frontend vip)

[1] https://github.com/openstack/puppet-tripleo/blob/master/manifests/haproxy.pp#L944-L972

**Comment by guits** [comment]
@dsavineau @fmount 
too many changes for supporting something that is even not supported by grafana itself afaik 😒

**Comment by fmount** [comment]
> @dsavineau @fmount
> too many changes for supporting something that is even not supported by grafana itself afaik unamused}

I agree on this, my concern is related to the fact that this patch can be (later) the cause of other bugs.
> 
> rhcs note: The more I look at this patch, the more I'm -1 to introduce such a change in a z-release

I see, and from a pure Ceph perspective, Grafana is used by the Ceph Dashboard: even though is
accessible, it's not exposed to the end user, so we can probably avoid this kind of change, which has
a big impact for a zstream.

**Comment by guits** [comment]
Grafana doesn't seem to offer a handy way for changing the admin password in a tool like ceph-ansible.
For all the reasons mentioned previously, I'm closing this PR.

As a workaround, a call with curl to the Grafana API can be made in order to change the password (+ reflecting the change by modifying `grafana_admin_password` in group_vars...)

**Pull request closed without merging**